lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerFactory.java

```diff
@@ -61,22 +64,24 @@ private DefaultRemoteManagerFactory(
     this.tpFactory = tpFactory;
   }
 
-  // TODO[REEF-547]: This method uses deprecated DefaultRemoteManagerImplementation constructor.
   @Override
   public RemoteManager getInstance(final String name) {
-    return new DefaultRemoteManagerImplementation(name,
-        DefaultRemoteManagerImplementation.UNKNOWN_HOST_NAME, // Indicate to use the localAddressProvider
-        0, // Indicate to use the tcpPortProvider
-        this.codec,
-        this.errorHandler,
-        this.orderingGuarantee,
-        this.numberOfTries,
-        this.retryTimeout,
-        this.localAddressProvider,
-        this.tpFactory);
+    try {
```

Comment:
Somewhat tangent to this PR, but this Factory class actually seems a bit suspect... It exposes parameters for us to inject at its creation, yet allows us to override those parameters explicitly when we call ``getInstance()``. From my searches, it seems like two ``getInstance()`` methods are only used in tests, one method only used in the Mesos runtime, and one method not used at all. @dongjoon-hyun what are your opinions here?